docs/proposals/20220330-topology-mutation-hook.md

```diff
@@ -58,12 +58,15 @@ Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/referen
 
 - **Inline patches**: are defined inline in a ClusterClass and implemented by the core CAPI controller.
 - **External patches**: are patches generated by an external component.
-- **Topology Mutation Hook**: is the hook defined in this proposal that allows users to plug in an external component that generates patches.
+- **Topology Mutation Hook**: is a hook defined in this proposal that allows users to plug in an external component that generates patches.
 - **External patch extension**: is an external component that generates patches.
+- **Inline variables**: are variables defined inline in a ClusterClass.
+- **External variables**: are varialbes defined by an external component.
+- **Variable Discovery Hook**: is a hook defined in this proposal that allows an external component to supply variable definitions.
 
 ## Summary
 
-This proposal introduces the Topology Mutation Hook, which makes it possible to mutate objects of the Cluster topology by generating patches externally. The patches are applied to templates defined in a ClusterClass.
+This proposal introduces the Topology Mutation Hook, which makes it possible to mutate objects of the Cluster topology by generating patches and associated variables externally. The patches and associated variables are applied to templates defined in a ClusterClass.
```

Comment:
I'm not sure the wording is good. We don't generate variables and we don't apply them to templates defined in a ClusterClass.

Maybe we should leave it out of the summary?

Reply:
Similar below in some cases